wrappers/wrapper.rb

```diff
@@ -835,17 +835,285 @@ def check_distances(vrp, unfeasible)
+          # the pause location but then we need to create alternative copies of these services
+          # for each vehicle, with different rest timewindows -- which kinda defeats the purpose.
+          #
+          # TODO: we could still simplify the other pauses of a vehicle even if some of the pauses cannot be
+          # simplified due to `max_service_duration > rest.tw` but this would complicate the post-processing
+          # e.g., we cannot shift everything later easily. At the moment, at most one pause is supported anyways.
+          next if vehicle.rests.any?{ |rest|
+            rest.timewindows.any? { |rest_tw|
+              rest_start = rest_tw.start || vehicle.timewindow&.start || 0
+              rest_end = rest_tw.end || vehicle.timewindow&.end || rest_start + vehicle.duration - rest.duration
+              max_service_duration > rest_end - rest_start
+            }
+          }
+
+          vehicle.rests.each{ |rest|
+            vehicle.duration -= rest.duration if vehicle.duration
+            vehicle.timewindow.end -= rest.duration if vehicle.timewindow&.end
+          }
+
+          vehicle[:simplified_rest_ids] = vehicle[:rest_ids].dup
+          vehicle[:rest_ids] = []
+          vehicle[:simplified_rests] = vehicle.rests.dup
+          vehicle.rests = []
+        }
+
+        vrp[:simplified_rests] = vrp.rests.select{ |r| vrp.vehicles.none?{ |v| v.rests.include?(r) } }
+        vrp.rests -= vrp[:simplified_rests]
+      when :rewind
+        # take the modifications back in case the vehicle is moved to another sub-problem
+        vrp.vehicles&.each{ |vehicle|
+          next unless vehicle[:simplified_rest_ids]&.any?
+
+          vehicle[:rest_ids].concat vehicle[:simplified_rest_ids]
+          vehicle[:simplified_rest_ids] = nil
+          vehicle.rests.concat vehicle[:simplified_rests]
+          vehicle[:simplified_rests] = nil
+
+          vehicle.rests.each{ |rest|
+            vehicle.duration += rest.duration if vehicle.duration
+            vehicle.timewindow.end += rest.duration if vehicle.timewindow&.end
+          }
+        }
+
+        if vrp[:simplified_rests]
+          vrp.rests.concat vrp[:simplified_rests]
+          vrp[:simplified_rests] = nil
+        end
+      when :patch_result
+        # correct the result with respect to simplifications
+        vrp.vehicles&.each{ |vehicle|
+          next unless vehicle[:simplified_rest_ids]&.any?
+
+          route = result[:routes].find{ |r| r[:vehicle_id] == vehicle.id }
+          no_cost = route[:activities].none?{ |a| a[:type] == 'service' }
+
+          # first shift every activity all the way to the left (earlier) if the route starts after
+          # the vehicle TW.start so that it is easier to do the insertions since there is no TW on
+          # services, we can do this even if force_start is false
+          shift_amount = vehicle.timewindow&.start.to_i - route[:start_time]
```

Comment:
Timewindows' start is 0 by default.

```suggestion
          shift_amount = vehicle.timewindow&.start - route[:start_time]
```

Reply:
if no `TW` we still want `0`